### ISSUE TITLE
Rawstrings get escaped like regular strings

### ISSUE THREAD

**Issue by da-kid** [rust-lang/rust#20431]
Hello,
I am using `rustc 0.13.0-nightly (10d99a973 2014-12-31 21:01:42 +0000)` an tried to compile the following Regex:
```rust
let regex = match Regex::new(r"([a-zA-Z\x{5b}-\x{60}\x{7b}-\x{7d}]+[a-zA-Z0-9\x{5b}-\x{60}\x{7b}-\x{7d}\-]+)") {
  Ok(re) => re,
  Err(err) => panic!("{}", err),
};
```
But as it seems, the ~~`\x{7d}` as well as~~ `\x7d` get escaped to a `}`.
This does also happen to other characters like `[`, as represented by a `\x5b`.
As far as I know, this is not supposed to happen, or at least it is counterintuitive, since it is a notation with a `r` prefix.

**Comment by huonw** [comment]
cc @burntsushi

**Comment by BurntSushi** [comment]
That regex does not compile because `\-` isn't a valid escape sequence. You can just use `-`.

Otherwise though, `\x{7d}` is part of the syntax of a regular expression. It has nothing to do raw strings. Raw strings are useful for writing regexes because you no longer have to deal with Rust's escaping rules *in addition* to regex's escaping rules. If they weren't escaped, I don't think your use of them here is valid anyway (in a character class).

**Comment by BurntSushi** [comment]
I think this should be reported on the regex repo? https://github.com/rust-lang/regex

**Comment by da-kid** [comment]
That is a workaround, but it should not be allowed normally.

**Comment by da-kid** [comment]
I just noticed that the rawstring is getting escaped when using `\x5b` instead of `\x{5b}`.
Therefor I am reopening this issue.

**Comment by BurntSushi** [comment]
Yes, that is part of the syntax. Have a look here: http://doc.rust-lang.org/regex/regex/index.html#escape-sequences

**Comment by da-kid** [comment]
@BurntSushi Oh, I see, what went wrong there:
I accidentally copied a private-use unicode character into the regex that was not displayed, but broke the parsing. Sorry for the distraction.
